Modules/FindBoost.cmake

```diff
@@ -862,6 +863,37 @@ function(_Boost_MISSING_DEPENDENCIES componentvar extravar)
 endfunction()
 
 #
+# Update library search directory hint variable with paths used by prebuilt boost binaries.
+#
+# Prebuilt windows binaries (https://sourceforge.net/projects/boost/files/boost-binaries/)
+# have library directories named using MSVC compiler version and architecture.
+# This function would append corresponding directories if MSVS is a current compiler,
+# so having Boost_ROOT would be enough to specify to find everything.
+#
+macro(_Boost_UPDATE_LIBRARY_SEARCH_DIRS_WITH_PREBUILD_PATHS componentlibvar basedir)
+  if(MSVC)
+    if(CMAKE_CL_64)
+      set(_arch_suffix 64)
+    else()
+      set(_arch_suffix 32)
+    endif()
+    if(MSVC80)
```

Comment:
Please check [CMAKE_CXX_COMPILER_ID](https://cmake.org/cmake/help/v3.6/variable/CMAKE_LANG_COMPILER_ID.html), [CMAKE_CXX_COMPILER_VERSION](https://cmake.org/cmake/help/v3.6/variable/CMAKE_LANG_COMPILER_VERSION.html), and [CMAKE_SIZEOF_VOID_P](https://cmake.org/cmake/help/v3.6/variable/CMAKE_SIZEOF_VOID_P.html) instead of the `MSVC*` and `CMAKE_CL_64` variables.  One may be able to compute the versioned directory name from `CMAKE_CXX_COMPILER_VERSION` instead of building a table that needs maintenance.